lib/tesla/middleware/telemetry.ex

```diff
@@ -13,21 +13,101 @@ if Code.ensure_loaded?(:telemetry) do
 
     end
 
-    :telemetry.attach("my-tesla-telemetry", [:tesla, :request], fn event, time, meta, config ->
+    :telemetry.attach("my-tesla-telemetry", [:tesla, :request, :stop], fn event, measurements, meta, config ->
       # Do something with the event
     end)
     ```
 
+    ## Telemetry Events
+
+    * `[:tesla, :request, :start]` - emitted at the beginning of the request.
+      * Measurement: `%{time: System.monotonic_time}`
+      * Metadata: `%{env: Tesla.Env.t()}`
+
+    * `[:tesla, :request, :stop]` - emitted at the end of the request.
+      * Measurement: `%{duration: native_time}`
+      * Metadata: `%{env: Tesla.Env.t()} | %{env: Tesla.Env.t, error: term}`
+
+    * `[:tesla, :request, :fail]` - emitted when there is an error.
```

Comment:
Sorry but I just proposed for us to name this `:failure` instead of `:fail`. I noticed `:fail` reads a bit weird when implementing the same on Broadway. Please let me know your thoughts. :)